packages/proxy/utils/tests.ts

```diff
@@ -35,18 +35,17 @@ export function createResponseStream(): [
 }
 
 export function createHeaderHandlers() {
-  const headers: Record<string, string> = {};
-  let statusCode = 200;
-
-  const setHeader = (name: string, value: string) => {
-    headers[name] = value;
-  };
-
-  const setStatusCode = (code: number) => {
-    statusCode = code;
+  const ref = {
+    headers: {} as Record<string, string>,
+    statusCode: -1,
+    setHeader(name: string, value: string) {
+      ref.headers[name] = value;
+    },
+    setStatusCode(code: number) {
+      ref.statusCode = code;
```

Comment:
I found this wasn't getting updated. Using an object in JS and passing it around is as good as passing with reference.